packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2617,19 +2662,31 @@ export class KeyringController extends BaseController<
 
   /**
    * Execute the given function after acquiring the controller lock
-   * and save the keyrings to state after it, or rollback to their
+   * and save the vault to state after it (only if needed), or rollback to their
    * previous state in case of error.
    *
    * @param callback - The function to execute.
+   * @param options - Options.
+   * @param options.forceUpdate - Force the vault update.
    * @returns The result of the function.
    */
   async #persistOrRollback<Result>(
     callback: MutuallyExclusiveCallback<Result>,
+    { forceUpdate }: { forceUpdate: boolean } = { forceUpdate: false },
```

Comment:
nit: Probably it's just me, but I feel like passing `true` to an attribute called `forceUpdate` feels wrong :sweat_smile: Thoughts about something like `alwaysUpdate`?

Also, since we are adding this just while we don't remove the deprecated overload from `withKeyring`, should we add another overload to `#persistOrRollback` with this option and deprecate that as well?